src/UIControls/MenuPaging.ts

```diff
@@ -0,0 +1,145 @@
+import { basePath, sgWorld } from "../Axiom";
+import { Button } from "../Button";
+import { Quaternion } from "../math/quaternion";
+import { Vector } from "../math/vector";
+import { radsToDegs } from "../Mathematics";
+import { Menu } from "../Menu";
+import { GetDeviceType, DeviceType, ProgramManager } from "../ProgramManager";
+
+
+export class MenuPaging extends Menu {
+
+  // override so we have a fixed size
+  cols: number = 3;
+  rows: number = 3;
+  buttonSize: number = 0.1;
+
+  // for paging
+  private pageNumber: number = 0;
+  private totalPages: number = 0;
+  private btnPL!: Button;
+  private btnPR!: Button;
+
+  constructor(public width: number, public height: number, public anchor: Vector<3>, public orientation: Quaternion, public anchorPosition: [number, number], public topAligned: boolean, public leftAligned: boolean, public horizontal: boolean) {
+    super(width, height, anchor, orientation, anchorPosition, topAligned, leftAligned, horizontal);
+
+    // recalculate the anchor so that it is centred. @Ruben better way to do this in the base class as an option
+    // @ruben better way to do this?
+    const newX = anchor.data[0] - ((this.cols * this.buttonSize) / 2);
+    anchor.data[0] = newX
+
+    this.corner = anchor.Copy().Sub(orientation.Apply(new Vector([width * (anchorPosition[0] - (leftAligned ? 0 : 1)), 0, height * (anchorPosition[1] - (topAligned ? 1 : 0))])));
+    console.log(`corner ${this.corner.data}`);
+    console.log(`xDirection ${this.xDirection.data}`);
+    console.log(`yDirection ${this.yDirection.data}`);
+
+    this.addPagingButtons();
+  }
+
+  /**
+   * Adds all the buttons as an array at once
+   *
+   * @param {Button[]} buttons
+   * @memberof MenuPaging
+   */
+  addButtons(buttons: Button[]) {
+    this.recomputeButtons = false; // stop recompute while we destroy the old ones
+    this.buttons.forEach(b => b.destroy());
+    this.buttons = [];
+    this.buttons.push(...buttons);
+    this.totalPages = Math.ceil(this.buttons.length / (this.cols * this.rows));
+    this.recomputeButtons = true;
+    this.Draw();
+  }
+  
+
+  private addPagingButtons() {
+    // add the paging buttons to the left and the right
+    const ypr = this.orientation.GetYPR();
+    const newPos = this.getNthButtonPosition(3); // @ruben I fudged this using your code for the 5th button. I'm sure you can do this a better way
```

Comment:
I would suggest using the `getButtonPosition` method instead so you can specify `x = -1 + 0.5` or `x = cols + 0.5` and `y = rows / 2`